File: OstjApp/OstjApi/Models/JobTitle.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace OstjApi.Models
{
    public class JobTitle
    {
        public int Id { get; set; }
        public int ProfileId { get; set; } = 0;
        [MaxLength(200)]
        public required string Title { get; set; }
#pragma warning disable CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
        public JobTitleDetails JobTitleDetails { get; set; }
#pragma warning restore CS8618 // Non-nullable field must contain a non-null value when exiting constructor. Consider adding the 'required' modifier or declaring as nullable.
    }

    public class JobTitleDetails
    {
        public int Id { get; set; }
        public int ProfileId { get; set; } = 0;
        [MaxLength(200)]
        public required string Title { get; set; }
        [Column(TypeName = "vector(4096)")]
        public Pgvector.Vector? Embedding { get; set; }
        public required bool IsUserDefined { get; set; } = false;
    }
}

File: backend/data-access/src/main/java/com/ostj/entities/Job_title.java
package com.ostj.entities;

public class Job_title {
    public int id = -1;
    public int profile_id = -1;
    public boolean is_user_defined;
    public String title;

    public String toString(){
        return String.format("Id=%d, profile_id=%s, Title=%s, is_user_defined=%d", id, profile_id, title, is_user_defined );
    }
}
